101-data-factory-hive-transformation/azuredeploy.json

```diff
@@ -0,0 +1,207 @@
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('azureStorageLinkedServiceName')]"
+            ],
+            "apiVersion": "2015-10-01",
+            "properties": {
+              "type": "AzureBlob",
+              "linkedServiceName": "[variables('azureStorageLinkedServiceName')]",
+              "typeProperties": {
+                "fileName": "[parameters('inputBlobName')]",
+                "folderPath": "[concat(parameters('blobContainer'), '/', parameters('inputBlobFolder'))]",
+                "format": {
+                  "type": "TextFormat",
+                  "columnDelimiter": ","
+                }
+              },
+              "availability": {
+                "frequency": "Month",
+                "interval": 1
+              },
+              "external": true
+            }
+          },
+          {
+            "type": "datasets",
+            "name": "[variables('blobOutputDatasetName')]",
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('azureStorageLinkedServiceName')]"
+            ],
+            "apiVersion": "2015-10-01",
+            "properties": {
+              "type": "AzureBlob",
+              "linkedServiceName": "[variables('azureStorageLinkedServiceName')]",
+              "typeProperties": {
+                "folderPath": "[concat(parameters('blobContainer'), '/', parameters('outputBlobFolder'))]",
+                "format": {
+                  "type": "TextFormat",
+                  "columnDelimiter": ","
+                }
+              },
+              "availability": {
+                "frequency": "Month",
+                "interval": 1
+              }
+            }
+          },
+          {
+            "type": "datapipelines",
+            "name": "[variables('pipelineName')]",
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('azureStorageLinkedServiceName')]",
+              "[variables('hdInsightOnDemandLinkedServiceName')]",
+              "[variables('blobInputDatasetName')]",
+              "[variables('blobOutputDatasetName')]"
+            ],
+            "apiVersion": "2015-10-01",
+            "properties": {
+              "description": "Pipeline that transforms data using Hive script.",
+              "activities": [
+                {
+                  "type": "HDInsightHive",
+                  "typeProperties": {
+                    "scriptPath": "[concat(parameters('blobContainer'), '/', parameters('hiveScriptFolder'), '/', parameters('hiveScriptFile'))]",
+                    "scriptLinkedService": "[variables('azureStorageLinkedServiceName')]",
+                    "defines": {
+                      "inputtable": "[concat('wasb://', parameters('blobContainer'), '@', parameters('storageAccountName'), '.blob.core.windows.net/', parameters('inputBlobFolder'))]",
```

Comment:
"inputtable" should be evaluated to: 
"inputtable": "https://changemeazurestorage.blob.core.windows.net/adfgetstarted/inputdata",
"partitionedtable": "https://changemeazurestorage.blob.core.windows.net/adfgetstarted/partitioneddata"

where changemeazurestorage is the name of the storage. Users change this name in the parameter file. As a prerequisite to deploying the template, user has to create inputdata folder in the adfgetstarted container and upload the input file (input.log) to the inputdata folder. 
The partitioneddata folder is automatically created when the Hive script runs. 

As per the best practice, I changed the above code to:
"inputtable": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2015-06-15').primaryEndpoints.blob,  parameters('blobContainer'), '/',parameters('inputBlobFolder'))]",
"partitionedtable": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2015-06-15').primaryEndpoints.blob,  parameters('blobContainer'), '/',parameters('outputBlobFolder'))]"

I can deploy the template locally from machine from Azure CLI successfully since the changemeazurestorage exists in the resource group I specify as a parameter. 

However, when I check in the code to GIT, the validation fails as the changemeazurestorage does not exist in the resource group that build uses. 

I appreciate if you can help resolve this issue.